Aux branch

gerkester opened the pull request.

gerkester committed:
Created Car3_Aux

Made modifications for specific use for aux team

gerkester committed:
re-added original Car3.py

John-Carr commented on Car3_Aux.py:
```diff
@@ -0,0 +1,30 @@
+emulator.CAN_init(baudRate=500e3)
+# Turn on 5v to power the CAN tranciver
+emulator.posSupply_init()
+mc = Mitsuba(0x08F89540, 0x01, emulator)
+mppt = Proton1(1024, 0x2, emulator)
+bms = Orion(0x6B0, 0x1, emulator)
```
Remove these because you removed the imports.

John-Carr commented on Car3_Aux.py:
```diff
@@ -0,0 +1,30 @@
+auxThread.daemon = True
+auxThread.start()
+
+bmsThread = threading.Timer(0.1, bms.sendCAN)
+bmsThread.daemon = True
+bmsThread.start()
```
same as above

John-Carr commented on Car3_Aux.py:
```diff
@@ -0,0 +1,30 @@
+
+try:
+    while True:
+        emulator.receiveCAN(cb)
```
Change this to just receive the Aux message so either above do a cb = {steering.receive msg} (check the mitsuba mock line 102:
```
def receiveCAN(self, data):
        print("Mitsuba Data: "+(" ".join("0x{:02x}".format(c) for c in data)))
```
And [this](https://www.geeksforgeeks.org/print-lists-in-python-4-different-ways/) on how to print lists in python.

gerkester replied:
I'll admit that I tried to look at the mitsuba file but it confused me more than anything. I wasn't sure what I was supposed to print a list for. So, I just assigned cb to a dictionary like it was in Car3.py. I tested that Car3_Aux runs successfully this time, but I can't do more than see if it runs because I don't have a DAD board.

John-Carr reviewed (changes requested):
Overall looks good! I put some comments on the changes in there and some documentation on how to print arrays in python. For that last comment I would add that to the aux steering class so we can see the data we receive.

gerkester committed:
Fixed Car3_Aux.py

Removed unnecessary references to deleted variables and modeled variable cb after original Car3.py

gerkester committed:
Created lights mock

Use Car3_AuxSteering.py for testing the steering board. Use Car3_Lights.py for testing the lights board.

John-Carr commented:
Looks like it should work have Jen test the branch with her stuff then merge it in :) @jenber23

jenber23 committed:
tweeked steering

jenber23 committed:
Merge branch 'auxBranch' of https://github.com/Solar-Gators/emulators into auxBranch
